From 1d4bb01a60548ae9a5439186fbe155f9f7d492e4 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Sun, 17 Nov 2019 18:45:49 +0100
Subject: Fix build with Qt 5.14

Summary:
std::hash<QString> is already specialized in Qt 5.14 [1], so we need
to avoid redifining it with this version.

[1] 4469e36d7203a55a4e158a50f0e9effc3f2fa3c2 in qtbase.git

Test Plan: Builds fine with Qt 5.14.0-beta3

Reviewers: #kde_pim

Subscribers: kde-pim

Tags: #kde_pim

Differential Revision: https://phabricator.kde.org/D25356
---
 src/shared/akstd.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/shared/akstd.h b/src/shared/akstd.h
index 4078ee4..774c432 100644
--- a/src/shared/akstd.h
+++ b/src/shared/akstd.h
@@ -29,6 +29,7 @@
 
 namespace std {
 
+#if (QT_VERSION < QT_VERSION_CHECK(5, 14, 0))
 template<>
 struct hash<QString> {
     using argument_type = QString;
@@ -39,6 +40,7 @@ struct hash<QString> {
         return qHash(s);
     }
 };
+#endif
 
 }
 
-- 
cgit v1.1
